animation/ProjectileAnimation.java

```diff
@@ -29,19 +31,26 @@ public int getOffsetY() {
     private int posX, posY;
     private int velocityX, velocityY;
     private boolean facingEast;
-    private Entity entity;
     private EffectType effectType;
-
-    public ProjectileAnimation(ProjectileAnimationType animationType, Entity entity) {
-        this.effectType = EffectType.ENTITY_EFFECT;
-        this.entity = entity;
+    private MapCollisionDetection mapCollision;
+    private ArrayList<Entity> targets;
+    private int damage;
+    private ArrayList<DamageMarker> targetMarkers;
+
+    public ProjectileAnimation(ProjectileAnimationType animationType, MapCollisionDetection collisionMap, Entity entity) {
+        this.effectType = EffectType.PROJECTILE_EFFECT;
+        this.mapCollision = collisionMap;
         this.facingEast = entity.getFacingEast();
+        this.targets = entity.getTargets();
         this.posX = entity.getPosX() + animationType.getOffsetX() * (entity.getFacingEast() ? 1 : -1);
         this.posY = entity.getPosY() + animationType.getOffsetY();
+        this.damage = entity.getDamage();
+        targetMarkers = new ArrayList<>();
         switch (animationType) {
             case YELLOW_FIRST:
                 setValues("arrow", 2);
                 velocityX = 10;
+				setNumLoops(500/velocityX);
```

Comment:
Technically it should be screen width/animation number/velocity, but that is more complicated than it's worth, alternatively int_max is also not a bad choice. This number just has to be big enough that you have enough loops so that your arrow animation doesn't stop after 2